### DISCUSSION TITLE
getServerSideProps or React Context API for shared data across page routes

### DISCUSSION THREAD

**Discussion by dkempton12** [vercel/next.js#30083] [Help]
Hello Next community,

I am working on an app with 3 page routes. The same data (nothing crazy, just rendering a handful of prices) will be used through the 3 pages of the app. 

When considering speed / performance, would I be better served to call `getServerSideProps` on each page to fetch this data, or store the data using the `React Context API`, wrapping `pages/_app.js` with my Provider and consuming the data within the 3 pages? 

Thank you for any information you're able to provide! I have researched this pretty thoroughly, and am unsure which approach to go with. Thanks!

**Comment by gaston-flores** [comment]
Are you really sure you will always need this data, for every page in your app? If you go for Context and `_app.js`, later on you may end up requesting data you don't really need. Unless it's something truly global to your app, it's better to let each page request its own data. 

If you are willing to wait for this data client-side, tools like [SWR](https://swr.vercel.app/) can cache the requests you make to avoid multiple calls for the same data.

**Comment by dkempton12** [comment]
Hi @gaston-flores thank you for your response. Yes this data will always be needed. The same data will be presented in a component on all three pages, based on the `productId` query parameter. In this scenario, would you say Context is the more suitable option?

**Reply by gaston-flores** [reply]
That should work just fine then. Just make sure to keep it in mind if your app ever grows beyond these three pages 😉
